### DISCUSSION TITLE
Trying to use this with gym stable retro (error).

### DISCUSSION THREAD

**Discussion by Disastorm** [RLE-Foundation/rllte#20] [Q&A] [answered]
I took some code I had that ran stable baselines 3 ppo with gym retro and tried to substitute this in and tried DAAC. However, I get the following error:

Calculated padded input size per channel: (160 x 1). Kernel size: (8 x 8). Kernel size can't be greater than actual input size
the shape of the observation is (height x width x channels) in this case 1. When I use frame stack of 4, the error becomes:
Calculated padded input size per channel: (160 x 4). Kernel size: (8 x 8). Kernel size can't be greater than actual input size

This is the observation space: Box(0, 255, (112, 160, 4), uint8)

This error is at 
rllte\xploit\encoder\mnih_cnn_encoder.py", line 65, in __init__
    n_flatten = self.trunk(sample.unsqueeze(0)).shape[1]

in your project, and in torch itself, the error is at
torch\nn\modules\conv.py", line 459, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,

What is this expecting?

Also I'm using stable baselines3 vec environments, SubprocVecEnv and the frame stack one.
I tried the gymnasium asyncvectorenv but they dont work with multiple retro environments i think.

**Comment by friedrichyuan** [comment, marked as answer]
The main reason for this error is that your input size (160, 4) is smaller than the kernel size (8, 8), we recommend you to use **PathakCnnEncoder** with the kernel size of 3.

``` python
from rllte.xploit.encoder import PathakCnnEncoder

...
encoder = PathakCnnEncoder()
agent.set(encoder)
...
```

Moreover, we will make many adjustments recently (still under development), so you are recommended to install the package with `git` rather than `pip`.

Thank you!

**Reply by Disastorm** [reply]
Thanks, I'll try that out.

**Reply by Disastorm** [reply]
Hello, can you tell me how to resume training from an agent.pth file that was trained using an RLLTE algorithm like DAAC?

**Reply by friedrichyuan** [reply]
Please refer to [Pre-training](https://docs.rllte.dev/tutorials/pre-training)

**Reply by Disastorm** [reply]
Ok I see thanks.
Also I'm not sure if I did something wrong but when I load the agent using th.load and then tell it to process an obs by using agent(obs)
the returned tensor seems to be the correct array size but it uses float values instead of ints, even though the original environment I trained it on was MultiBinary.
I even just checked the env shape in the training and it was multibinary, although after it passes through the asyncvector environment, it becomes a Box(0, 1, (8, 12), int8) 

But since that is still specifying int rather than float it still looks ok I think, so I'm not sure why my trained agent is returning floats.

Could it be because I didn't do any pre-training?

**Reply by friedrichyuan** [reply]
Can you open an issue so that we can track your problem in detail?

Since the project is still under development, it can help us to improve it.

You're also welcome to make contributions. Thanks!

**Reply by Disastorm** [reply]
Ok thanks, I wasn't sure if i did something wrong or not, i'll make an issue then.
